tests/models/test_cat_model.py

```diff
@@ -297,3 +297,66 @@ async def test_find_many(
     )
 
     assert found_cat_summaries == expected_cat_summaries
+
+
+@pytest.mark.parametrize(
+    "existing_cat_documents, cat_id, expected_result",
+    [
+        (
+            [
+                {
+                    "_id": ObjectId("000000000000000000000101"),
+                    "name": "Sammybridge Cat",
+                    "ctime": datetime(2020, 1, 1, 0, 0, tzinfo=UTC),
+                    "mtime": datetime(2020, 1, 1, 0, 0, tzinfo=UTC),
+                    "is_removed": False,
+                },
+                {
+                    "_id": ObjectId("000000000000000000000102"),
+                    "name": "Shirasu Sleep Industries Cat",
+                    "ctime": datetime(2020, 1, 1, 0, 0, tzinfo=UTC),
+                    "mtime": datetime(2020, 1, 1, 0, 0, tzinfo=UTC),
+                    "is_removed": False,
+                },
+            ],
+            dto.CatID("000000000000000000000101"),
+            dto.UpdatedCount(
+                count=1,
+            ),
+        ),
+        (
+            [
+                {
+                    "_id": ObjectId("000000000000000000000101"),
+                    "name": "Sammybridge Cat",
+                    "ctime": datetime(2020, 1, 1, 0, 0, tzinfo=UTC),
+                    "mtime": datetime(2020, 1, 1, 0, 0, tzinfo=UTC),
+                    "is_removed": False,
+                },
+                {
+                    "_id": ObjectId("000000000000000000000102"),
+                    "name": "Shirasu Sleep Industries Cat",
+                    "ctime": datetime(2020, 1, 1, 0, 0, tzinfo=UTC),
+                    "mtime": datetime(2020, 1, 1, 0, 0, tzinfo=UTC),
+                    "is_removed": False,
+                },
+            ],
+            dto.CatID("000000000000000000000000"),
+            dto.UpdatedCount(
+                count=0,
+            ),
+        ),
+    ],
+)
+@conftest.async_test
+async def test_delete_one(
+    existing_cat_documents: List[BSONDocument],
+    cat_id: dto.CatID,
+    expected_result: dto.UpdatedCount,
+) -> None:
+    collection = await get_collection(cat_model._COLLECTION_NAME)
+    await collection.insert_many(existing_cat_documents)
+
+    found_cat = await cat_model.delete_one(cat_id)
+
+    assert found_cat == expected_result
```

Comment:
[suggestion] We probably should also check if what is stored in the DB is what as expect (so you need to retrieve the document and check if that document is what we expect it to be)

Reply:
done